source/server/options_impl.h

```diff
@@ -71,6 +71,9 @@ class OptionsImpl : public Server::Options {
   void setHotRestartDisabled(bool hot_restart_disabled) {
     hot_restart_disabled_ = hot_restart_disabled;
   }
+  void setSignalHandling(bool signal_handling_enabled) {
```

Comment:
Does this need a setter for your own use? Can you just pass the value in the constructor?

Reply:
I think a setter is more ergonomic to avoid constructor argument creep. WDYT?

Reply:
In general I think if someone is only ever going to do:
1) Construct
2) Set

It should just be in the constructor. But I don't feel that strongly about it if you feel otherwise.

Reply:
Understood. I want to keep it a setter and not change the type signature if I don't have to.

Reply:
OK

Reply:
Oops sorry. Do you have test coverage for this option?

Reply:
No, good catch. I will write a test in OptionsImpl or maybe main_common_test and make sure this ignores signals properly. Will ping again Monday.

Reply:
I put a setter test into `options_impl_test` in db302c7. This is idiomatic but doesn't really test the full cycle of startup -> `main_common_base.server().options().setFoo()` -> `.run()`. Let me know if it's enough.